lib/AST/ASTContext.cpp

```diff
@@ -1642,6 +1642,21 @@ void ASTContext::addModuleInterfaceChecker(
   getImpl().InterfaceChecker = std::move(checker);
 }
 
+void ASTContext::setModuleAliases(const llvm::StringMap<StringRef> &aliasMap) {
+  for (auto k: aliasMap.keys()) {
+    auto val = aliasMap.lookup(k);
+    ModuleAliasMap[getIdentifier(k)] = getIdentifier(val);
+  }
+}
+
+Identifier ASTContext::lookupModuleAlias(Identifier key) const {
+  auto found = ModuleAliasMap.find(key);
+  if (found != ModuleAliasMap.end()) {
+    return found->second;
+  }
+  return Identifier();
```

Comment:
Would it be more useful to return `key` instead of an empty value? (I haven't looked at the use sites for this, so I don't know the answer to that question.)

Reply:
Good question, either check will work, but returning key will incur less cost, so will do that then :)